lib/core/include/irods/transport/default_transport.hpp

```diff
@@ -1,5 +1,16 @@
-#ifndef IRODS_IO_DEFAULT_TRANSPORT_HPP
-#define IRODS_IO_DEFAULT_TRANSPORT_HPP
+#undef IRODS_IO_DEFAULT_TRANSPORT_HPP_INCLUDE_HEADER
+
+#if defined(IRODS_IO_TRANSPORT_ENABLE_SERVER_SIDE_API)
+#  if !defined(IRODS_IO_DEFAULT_TRANSPORT_HPP_FOR_SERVER)
+#    define IRODS_IO_DEFAULT_TRANSPORT_HPP_FOR_SERVER
+#    define IRODS_IO_DEFAULT_TRANSPORT_HPP_INCLUDE_HEADER
+#  endif
+#elif !defined(IRODS_IO_DEFAULT_TRANSPORT_HPP_FOR_CLIENT)
+#  define IRODS_IO_DEFAULT_TRANSPORT_HPP_FOR_CLIENT
+#  define IRODS_IO_DEFAULT_TRANSPORT_HPP_INCLUDE_HEADER
+#endif
+
+#ifdef IRODS_IO_DEFAULT_TRANSPORT_HPP_INCLUDE_HEADER
```

Comment:
So, in order to include both would be like this...?
```c++
#include <irods/transport/default_transport.hpp> // client-side

#undef IRODS_IO_DEFAULT_TRANSPORT_HPP_INCLUDE_HEADER

#define IRODS_IO_TRANSPORT_ENABLE_SERVER_SIDE_API
#include <irods/transport/default_transport.hpp> // server-side
```
But then... everything gets `#undef`'d on L15-21... and redefined. So, I guess that wouldn't work.

How do you do it? :p

Reply:
D'oh, it gets `undef`'d in this file. So I guess you would need `IRODS_IO_DEFAULT_TRANSPORT_HPP_FOR_SERVER` in there somewhere.

Reply:
For the default transport:
```cpp
#include <irods/transport/default_transport.hpp>

#define IRODS_IO_TRANSPORT_ENABLE_SERVER_SIDE_API
#include <irods/transport/default_transport.hpp>
```
For filesystem:
```cpp
#include <irods/filesystem.hpp>

#define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
#include <irods/filesystem.hpp>
```